### ISSUE TITLE
Error 'squiggles' don't clear in JSON files.

### ISSUE THREAD

**Issue by pedro-w** [Togusa09/vscode-tmlanguage#19]
If I edit a non-tmLanguage JSON file when this extension is enabled, it throws up a lot of problem reports which don't get cleared when I continue typing. For example if I am trying to  type `{ "size" : 0 }` I see a series of problems for each of the intermediate steps which don't cleared. Without this extension active they do get cleared when I press the 0 (i.e. when it becomes valid JSON). See attached picture.
```
Version: 1.27.2 (user setup)
Commit: f46c4c469d6e6d8c46f268d1553c5dc4b475840f
Date: 2018-09-12T16:17:45.060Z
Electron: 2.0.7
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64
Textmate Languages: 0.9.0
```
Note this is not a textmate file, so this extension probably should not have been activated (as reported in #17 )
![not-clearing-errors](https://user-images.githubusercontent.com/22172519/45691941-843aef00-bb51-11e8-8b92-02f3226e4071.png)

**Comment by IronGeek** [comment]
I've reported the problem with JSON errors not being cleared too, here: #14...

**Comment by pedro-w** [comment]
@IronGeek you're right, sorry I missed that. Should I close this one as a duplicate?

**Comment by IronGeek** [comment]
@pedro-w I think you should keep this one open for visibility. The other issue (#14) doesn't specifically mention anything about `JSON` in the title, so it's pretty easy to miss over this one...

**Comment by pedro-w** [comment]
OK, I think I've got a fix for this. I will put in a Pull Request hopefully later today.
